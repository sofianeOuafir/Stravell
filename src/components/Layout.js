import React from "react";
import Head from "next/head";

import Footer from './Footer';
import Header from './Header';

const Layout = ({ withTitleAndDescription = true, title, description, children }) => (
  <div>
    {withTitleAndDescription && (
      <Head>
        <title>{title}</title>
        <meta name="description" content={description} />
      </Head>
    )}

    <Header />
    <div className="header__margin-top">
      {children}
    </div>
    
    <Footer />
  </div>
);

export default Layout;
