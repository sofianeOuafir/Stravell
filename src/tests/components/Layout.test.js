import React from "react";
import { shallow } from "enzyme";
import Layout from "./../../components/Layout";

describe('a title and description are provided', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<Layout title="my title" description="my description"><p>hello</p></Layout>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('not title and description are provided', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<Layout withTitleAndDescription={false}><p>hello</p></Layout>);
    expect(wrapper).toMatchSnapshot();
  });
});

