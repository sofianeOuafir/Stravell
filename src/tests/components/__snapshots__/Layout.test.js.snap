// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a title and description are provided should render correctly 1`] = `
<div>
  <SideEffect(Head)>
    <title>
      my title
    </title>
    <meta
      content="my description"
      name="description"
    />
  </SideEffect(Head)>
  <Connect(Header) />
  <div
    className="header__margin-top"
  >
    <p>
      hello
    </p>
  </div>
  <Footer />
</div>
`;

exports[`not title and description are provided should render correctly 1`] = `
<div>
  <Connect(Header) />
  <div
    className="header__margin-top"
  >
    <p>
      hello
    </p>
  </div>
  <Footer />
</div>
`;
